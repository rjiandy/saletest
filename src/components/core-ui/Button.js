// @flow

import React from 'react';

type Props = {
    text: string;
    onClick?: () => void;
    style?: {[key: string]: any};
};

export default function Button(props: Props) {
    let {text, onClick, style} = props;
    let composedStyle = {...styles.buttonContainer, ...style};
    return (
        <div onClick={onClick} style={composedStyle}>
            <span style={styles.buttonText}>{text}</span>
        </div>
    )
}

const styles = {
    buttonContainer: {
        display: 'flex',
        width: '100px',
        height: '40px',
        cursor: 'pointer',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 2,
        backgroundColor: 'rgb(215, 60, 60)',
    },
    buttonText: {
        fontSize: 12,
        color: 'white',
        fontWeight: '700',
    },
}