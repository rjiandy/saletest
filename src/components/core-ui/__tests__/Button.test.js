import React from 'react';

import Button from '../Button';
import renderer from 'react-test-renderer';


it('renders correctly with props', () => {
    const component = renderer
      .create(<Button text="asd" onClick={() => null} style={{flex: 1}} />);

    expect(component.toJSON()).toMatchSnapshot();
    let tree = component.toJSON();

    tree.props.onClick();
    expect(tree).toMatchSnapshot();
  });