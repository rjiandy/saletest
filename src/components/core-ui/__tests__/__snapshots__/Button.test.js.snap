// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(215, 60, 60)",
      "borderRadius": 2,
      "cursor": "pointer",
      "display": "flex",
      "flex": 1,
      "height": "40px",
      "justifyContent": "center",
      "width": "100px",
    }
  }
>
  <span
    style={
      Object {
        "color": "white",
        "fontSize": 12,
        "fontWeight": "700",
      }
    }
  >
    asd
  </span>
</div>
`;

exports[`renders correctly with props 2`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(215, 60, 60)",
      "borderRadius": 2,
      "cursor": "pointer",
      "display": "flex",
      "flex": 1,
      "height": "40px",
      "justifyContent": "center",
      "width": "100px",
    }
  }
>
  <span
    style={
      Object {
        "color": "white",
        "fontSize": 12,
        "fontWeight": "700",
      }
    }
  >
    asd
  </span>
</div>
`;
